orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -66,31 +68,42 @@ def forward(self, *inputs, **kwargs):
-                                                      self._onnx_model,
-                                                      inputs,
-                                                      kwargs)
+        if self._fast_path == False:
+            input_info = _io.parse_inputs_for_onnx_export(self._module_parameters,
+                                                        self._onnx_model,
+                                                        inputs,
+                                                        kwargs)
 
-        # Reinitialize graph builder if the inputs or initializers requiring gradient have changed.
-        # Order of or operation is important here because we always need to call
-        # _reinitialize_graph_builder irrespective of the value of build_gradient_graph.
-        build_gradient_graph = self._reinitialize_graph_builder(input_info) or build_gradient_graph
+            # Reinitialize graph builder if the inputs or initializers requiring gradient have changed.
+            # Order of or operation is important here because we always need to call
+            # _reinitialize_graph_builder irrespective of the value of build_gradient_graph.
+            build_gradient_graph = self._reinitialize_graph_builder(input_info) or build_gradient_graph
 
         # Build the gradient graph
         if build_gradient_graph:
             self._build_graph()
 
-        device = _utils.get_device_from_module(self._original_module) or \
-            _utils.get_device_from_inputs(inputs, kwargs)
-        # The _training_session/_inference_session should be created every time
-        # the graph was built or if the device changed between calls to forward
-        create_execution_session = build_gradient_graph or self._device != device
-        if self._device != device:
-            self._device = device
+        create_execution_session = False
+        if self._fast_path == False:
+            device = _utils.get_device_from_module(self._original_module) or \
+                _utils.get_device_from_inputs(inputs, kwargs)
+            # The _training_session/_inference_session should be created every time
+            # the graph was built or if the device changed between calls to forward
+            create_execution_session = build_gradient_graph or self._device != device
+            if self._device != device:
+                self._device = device
+
         if create_execution_session:
             # Create execution session creates the training_session
             self._create_execution_agent()
 
+            if self._freeze:
+                # enabled fast path when execution session is created the first time
+                self._fast_path = True
+                if self._loglevel <= _logger.LogLevel.WARNING:
+                    warnings.warn("Fast path enabled, some checks will be skipped by assuming model's schema remains unchanged during training.",
```

Comment:
"schema" is not a public concept of ORTModule. We have to be explicit on what is expected not to change (number of inputs, number of outputs, device change for each tensor, etc)

Reply:
warning updated per skip list config names